public/account.php

```diff
@@ -1,20 +1,40 @@
-<?php require('header.php'); ?>
+<?php 
+require('header.php');
+require('init_form.php');
 
-<?php require('database.php'); ?>
+if ($_SESSION['user'] === NULL) {
+	header("Location: http://rdripley.com/forum");
+}
+
+$password = $_POST['password'];
+$userId= $_SESSION['user_id'];
+$new_password = $_POST['new_password'];
+
+if (formIsBeingSubmitted()) {
+	if ($_POST['confirm_password'] === $_POST['new_password'] && strlen($_POST['new_password']) >= 8) {
+	$db = new Database();
+	$result = $db->query("UPDATE users SET password= $_POST[new_password] WHERE id= $_SESSION[user_id]");
```

Comment:
To debug this, var_dump the exact string you're providing to `query`. The PHP variables should have been replaced by their values.

I assume the problem is that you didn't put any quotes outside of the array keys. I.e. You did `$_POST[new_password]` instead of `$_POST['new_password']`.